partner_preview_scripts/encore_parser.py

```diff
@@ -0,0 +1,610 @@
+            .withColumn("cellLineName", f.col("cellInfo").getItem(0))
+            .withColumn("experimentId", f.col("cellInfo").getItem(1))
+            .filter(f.col("zscore") != "NaN")
+            # BLISS data is not aggregated for cell lines, instead we have data for each replicates.
+            # To allow averaging, data needs to be grouped by gene pair and cell line:
+            .groupby("id", "cellLineName")
+            # Averaging z-scores:
+            .agg(
+                f.sum(f.col("zscore")).alias("sumzscore"),
+                f.count(f.col("zscore")).alias("experiment_count"),
+            )
+            .withColumn(
+                "geneticInteractionScore",
+                f.col("sumzscore") / f.sqrt(f.col("experiment_count")),
+            )
+            # Calculating p-value for the averaged z-score:
+            .withColumn(
+                "geneticInteractionPValue",
+                f.udf(
+                    lambda zscore: float(st.norm.sf(abs(zscore)) * 2)
+                    if zscore
+                    else None,
+                    t.FloatType(),
+                )(f.col("geneticInteractionScore")),
+            )
+            .select(
+                "id",
+                "cellLineName",
+                f.lit("bliss").alias("statisticalMethod"),
+                "geneticInteractionPValue",
+                "geneticInteractionScore",
+            )
+            # Based on the z-score we can tell the nature of the interaction:
+            # - positive z-score means synergictic
+            # - negative z-score means antagonistics interaction
+            .withColumn(
+                "geneInteractionType",
+                f.when(f.col("geneticInteractionScore") > 0, "synergistic").otherwise(
+                    "antagonistic"
+                ),
+            )
+            .persist()
+        )
+
+    def get_gemini_data(self, gemini_file: str) -> DataFrame:
+        """Parsing cooperative effects calculated by gemini method
+
+        The structure of the file similar to the lfc file, the process follows a similar fashion,
+        except it slightly different. Different enough to get a separate function.
+        """
+
+        # Fixed statistical field names:
+        stats_fields = ["score", "pval", "FDR"]
+
+        # Reading the data into a single dataframe:
+        gemini_df = self.spark.read.csv(gemini_file, sep="\t", header=True)
+
+        # Collect the cell lines from the lfc file header:
+        cell_lines = set(
+            [
+                "_".join(x.split("_")[:-1])
+                for x in gemini_df.columns[4:]
+                if x.startswith("SID")
+            ]
+        )
+
+        # There are some problems in joining gemini files on Encore side. It causes a serious issues:
+        # 1. Multiple Gene_Pair columns in the file -> these will be indexed in the pyspark dataframe
+        # 2. Some columns for some cell lines will be missing eg. pvalue for SIDM00049_CSID1053
```

Comment:
Yes, based on the logs, this is not an issue in the new release, but I don't want to remove the logic, because, who knows, there might be problem with it later.